Intune, PowerShell & Graph API/demo_graph.ps1

```diff
@@ -0,0 +1,158 @@
+#region Load Helper function
+. .\Connect_GraphAPI.ps1
+
+get-help Connect-GraphAPI
+#endregion
+
+#region Authenticate
+$credentials = (get-credential)
+$credentials.Password.MakeReadOnly()
+
+connect-azuread -Credential $credentials
+
+Get-AzureADTenantDetail
+$tenantID = (Get-AzureADTenantDetail).ObjectID
+
+Get-AzureADApplication
+$appID = (Get-AzureADApplication | where {$_.displayname -like '*Intune Automation*'}).AppID
+
+$GraphCredentials = @{
+    "AzureCredentials" = $credentials
+    "TenantID" = $tenantID
+    "AppID" = $appID
+}
+
+Connect-GraphAPI @GraphCredentials
+#endregion
+
+#region check header
+$authHeader = Connect-GraphAPI @GraphCredentials
+$authHeader
+$authHeader.ExpiresOn.LocalDateTime
+#endregion
+
+#region set some variables
+$graphApiVersion = "Beta"
+$graphURI = "https://graph.microsoft.com/$graphapiversion"
+$graphURI
+#endregion
+
+#region get user info
+$Resource = "/users"
+$uri = $graphURI + $Resource
+$uri
+
+Invoke-RestMethod -Uri $uri -Method GET -Headers $authHeader
+
+
+$users = (Invoke-RestMethod -Uri $uri -Method GET -Headers $authHeader).value
+$users
+$users | select-object DisplayName, id
+$userid = ($users | where {$_.displayname -like 'Ralph Eckhard'}).id
+$userid
+
+$uri
+$uri = $uri + '/' + $userid
+$uri
+
+Invoke-RestMethod -Uri $uri -Method GET -Headers $authHeader
+$UserRalph = (Invoke-RestMethod -Uri $uri -Method GET -Headers $authHeader)
+$UserRalph.Displayname
+$UserRalph | Select-Object DisplayName, MobilePhone, City
+
+$PatchJSON = @{
+  "mobilephone" = "+31640409642"
```

Comment:
misschien deze scrubben? :)

Reply:
Nah, is m'n zakelijke nummer. Vind ik wel prima :)